src/Main.elm

```diff
@@ -1,37 +1,143 @@
-module Main exposing (Model, Msg(..), book, init, main, update, view)
+module Main exposing (Model, Msg(..), init, main, update, view)
 
 import BookData exposing (Book, allBooks)
 import Browser
-import Html exposing (Html, div, h1, img, li, text, ul)
+import Html exposing (Html, button, div, h1, img, li, table, td, text, th, thead, tr, ul)
 import Html.Attributes exposing (alt, class, src)
+import Html.Events exposing (onClick)
+import List exposing (append, filter, map, sum)
 
 
 
 ---- MODEL ----
 
 
 type alias Model =
-    { books : List Book }
+    { books : List Book, cart : Cart }
+
+
+type alias CartItem =
+    { id : BookID, quantity : Int }
+
+
+type alias Cart =
+    List CartItem
+
+
+type alias BookID =
+    Int
+
+
+add : Cart -> BookID -> Cart
+add cart bookID =
```

Comment:
Nitpick, but I think I'd reverse the order of arguments here, just semantically: add X to Y.

Ditto for `remove`, etc.

I'd also be tempted to name the function `addToCart`, just for clarity's sake. (+ again for `remove`, etc)